.github/workflows/pr-preview.yml

```diff
@@ -0,0 +1,102 @@
+name: PR Preview
+
+on:
+  push:
+    branches:
+      - '**'
+
+# Avoid duplicate workflows on same branch
+concurrency: 
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  pr-preview:
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        shell: bash -ileo pipefail {0}
+
+    steps:
+      - name: Checkout Streamlit code
+        uses: actions/checkout@v3
+        with: 
+          persist-credentials: false
+          submodules: 'recursive'
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+      - name: Setup virtual env
+        uses: ./.github/actions/make_init
+      - name: Create Wheel File
+        timeout-minutes: 120
+        run: |
+          sudo apt install rsync
+          BUILD_AS_FAST_AS_POSSIBLE=1 make package
+      # Uses action to safely process user input (branch name) to prevent script injection attacks
+      - name: Set Environment Variables
+        uses: ./.github/actions/preview_branch
+        with:
+          pull_request_number: ${{ github.event.pull_request.number }}
+          ref_type: ${{ github.ref_type }}
+          branch: ${{ github.ref_name }}
+
+      # TODO: Need the AWS S3 secret/key for the below steps - added the if statement to gate for now
+      - if: ${{ 'true' == 'false' }}
+        name: Upload wheel to S3
+        env:
+          BRANCH: ${{ env.BRANCH }}
+          PREVIEW_BRANCH: ${{ env.PREVIEW_BRANCH }}
+          CORE_PREVIEWS_S3_KEY_ID: ${{ secrets.CORE_PREVIEWS_S3_KEY_ID }}
+          CORE_PREVIEWS_S3_SECRET_KEY: ${{ secrets.CORE_PREVIEWS_S3_SECRET_KEY }}
+        run: |
+          sudo apt install -y awscli
+
+          aws configure set aws_access_key_id "$CORE_PREVIEWS_S3_KEY_ID"
+          aws configure set aws_secret_access_key "$CORE_PREVIEWS_S3_SECRET_KEY"
+
+          # GitHub upgraded ubuntu-latest environment from 18.04 to 20.04 which upgraded aws-cli
+          # from v1 to v2. When running in GH env, v2 of aws-cli throws an error if no region is specified.
+          aws configure set region us-west-2
+
+          cd lib/dist
+          export WHEELFILE="$(ls -t *.whl | head -n 1)"
+
+          if [ "${BRANCH}" = "release/demo" ]
+          then
+            aws s3 cp "${WHEELFILE}" s3://core-previews/${PREVIEW_BRANCH}/streamlit-11.11.11-py2.py3-none-any.whl --acl public-read
+            echo -e "Wheel file download link: https://core-previews.s3-us-west-2.amazonaws.com/${PREVIEW_BRANCH}/streamlit-11.11.11-py2.py3-none-any.whl"
```

Comment:
nit: technically not quite your change since it's being done this way in CircleCI, but it'd be nice to have a comment explaining the weird v11.11.11 version number here so that people don't get confused as to why we're doing this down the line